urllib3/connectionpool.py

```diff
@@ -568,25 +590,19 @@ def urlopen(self, method, url, body=None, headers=None, retries=None,
             # Close the connection. If a connection is reused on which there
             # was a Certificate error, the next request will certainly raise
             # another Certificate error.
-            if conn:
-                conn.close()
-                conn = None
+            conn = self._cleanup_conn(conn, not release_conn)
```

Comment:
I agree. release_conn though is a good name for what we want to do explicitly in _cleanup_conn, but we only want to do it if we're not releasing it in the calling method. It just so happens that it's also a good name in the calling method since that's what he's going to do. I could change it to something like self._cleanup_conn(release_conn == False), otherwise we could do something like:

def _cleanup_conn(self, conn, auto_release):
   ....
   if not auto_release:
        self._put_conn(None)

...and have the caller do self._cleanup_conn(conn, release_conn).
I'm open to suggestions....